component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpWorkerThread.java

```diff
@@ -18,62 +18,64 @@
 
+import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.stream.Collectors;
 
 /**
  * Handles the send data to source listener.
  */
 public class HttpWorkerThread implements Runnable {
     private static final Logger logger = LoggerFactory.getLogger(HttpWorkerThread.class);
-    private CarbonMessage carbonMessage;
-    private CarbonCallback carbonCallback;
+    private HTTPCarbonMessage carbonMessage;
     private SourceEventListener sourceEventListener;
     private String sourceID;
     private String[] trpProperties;
 
-    HttpWorkerThread(CarbonMessage cMessage, CarbonCallback cCallback, SourceEventListener sourceEventListener,
+    HttpWorkerThread(HTTPCarbonMessage cMessage, SourceEventListener sourceEventListener,
                      String sourceID, String[] trpProperties) {
         this.carbonMessage = cMessage;
-        this.carbonCallback = cCallback;
         this.sourceEventListener = sourceEventListener;
         this.sourceID = sourceID;
         this.trpProperties = trpProperties;
     }
 
     @Override
     public void run() {
+        BufferedReader buf = new BufferedReader(
+                new InputStreamReader(new HttpMessageDataStreamer(carbonMessage).getInputStream(),
+                        Charset.defaultCharset()));
         try {
-            InputStream inputStream = carbonMessage.getInputStream();
-            String payload = new String(ByteStreams.toByteArray(inputStream), StandardCharsets.UTF_8);
+            String payload = buf.lines().collect(Collectors.joining("\n"));
+
             if (!payload.equals(HttpConstants.EMPTY_STRING)) {
                 sourceEventListener.onEvent(payload, trpProperties);
-                HttpSourceUtil.handleCallback("OK", carbonCallback, 200);
+                HttpSourceUtil.handleCallback(carbonMessage, 200);
                 if (logger.isDebugEnabled()) {
                     logger.debug("Submitted Event " + payload + " Stream");
                 }
             } else {
-                HttpSourceUtil.handleCallback("Empty payload event", carbonCallback, 405);
+                HttpSourceUtil.handleCallback(carbonMessage, 405);
                 if (logger.isDebugEnabled()) {
                     logger.debug("Empty payload event, hence dropping the event chunk in " + sourceID);
                 }
             }
-        } catch (IOException e) {
-            logger.error("Event format is not equal to the expected in " + sourceID);
-            HttpSourceUtil.handleCallback("Event format " +
-                    "is not equal to the expected", carbonCallback, 405);
         } finally {
-            carbonMessage.release();
+            try {
+                buf.close();
+            } catch (IOException e) {
+                logger.error("Error closing byte buf");
```

Comment:
Log e as well

Reply:
Fixed in 6bf095af9d7a4c00220069badb50517cec553139